src/SingleTask.tsx

```diff
@@ -3,21 +3,74 @@ import { Task } from './types'
 
 type SingleTaskProp = {
   singleTask: Task
-  togglecheck: (toggleCheck: Task) => void
+  updatedTask: (toggleCheck: Task) => void
 }
 
-const SingleTask: React.FC<SingleTaskProp> = ({ singleTask, togglecheck }: SingleTaskProp) => {
+const SingleTask: React.FC<SingleTaskProp> = ({ singleTask, updatedTask }: SingleTaskProp) => {
   const toggleCheck = (): void => {
-  
-       singleTask.checkedStatus = !singleTask.checkedStatus
+    singleTask.checkedStatus = !singleTask.checkedStatus
+    updatedTask(singleTask)
+  }
+
+  const [editTaskButtonClick, setEditTaskButtonClick] = React.useState<boolean>(false)
+  const [inputValueOfEditedTask, setInputValueOfEditedTask] = React.useState<string>(singleTask.title)
 
-    togglecheck(singleTask)
+  const handleEditingTaskInput = (event : React.ChangeEvent<HTMLInputElement>) => {
+    setInputValueOfEditedTask(event.target.value)
   }
 
-    return (
+  const handleEditedTaskInput = () => {
+    editMainState()
+  }
+
+  const handleKeypress = (event) => {
+    if (event.keyCode === 13) {
+      handleEditedTaskInput()
+    } else {
+      handleEditingTaskInput(event)
+    }
+  }
+
+  const editMainState = () => {
+    singleTask.title = inputValueOfEditedTask
+    updatedTask(singleTask)
+    setEditTaskButtonClick(false)
+  }
+
+  // const DeletingTask = () => {
+  //   delete singleTask
+  // }
+
+  return (
     <div>
-      <input type="checkbox" checked={singleTask.checkedStatus} onChange={() => toggleCheck()} />
-      <span className={singleTask.checkedStatus ? 'background-red' : ''}>{singleTask.title}</span>
+      {editTaskButtonClick ? (
+        <div>
+          <input
+            type="text"
+            value={inputValueOfEditedTask}
+            onChange={handleEditingTaskInput}
```

Comment:
since there is just one line in handleEditingTaskInput we can move the code in this line 
{(event) => setInputValueOfEditedTask(event.target.value)}